src/dialects/abstract/query-interface-typescript.ts

```diff
@@ -0,0 +1,34 @@
+import type { QueryRawOptions, Sequelize } from '../../sequelize';
+import type { AbstractQueryGenerator } from './query-generator';
+
+export interface QueryInterfaceOptions {
+  sequelize: Sequelize;
+  queryGenerator: AbstractQueryGenerator;
+}
+
+export class AbstractQueryInterfaceTypeScript {
+  protected readonly sequelize: Sequelize;
+  protected readonly queryGenerator: AbstractQueryGenerator;
+
+  constructor(options: QueryInterfaceOptions) {
+    if (!options.sequelize) {
+      throw new Error('QueryInterface initialized without options.sequelize');
+    }
+
+    if (!options.queryGenerator) {
+      throw new Error('QueryInterface initialized without options.queryGenerator');
```

Comment:
same here about checking being redundant with typescript

Reply:
We are doing this in the `AbstractQueryGeneratorTypeScript` as well